spec/access/AccessControl.behavior.ts

```diff
@@ -0,0 +1,169 @@
+        expect(
+          await instance.callStatic['getRoleAdmin(bytes32)'](
+            DEFAULT_ADMIN_ROLE,
+          ),
+        ).to.equal(DEFAULT_ADMIN_ROLE);
+      });
+    });
+    describe('#grantRole(bytes32, address)', function () {
+      it('accounts can be granted a role multiple times', async function () {
+        await instance.connect(admin).grantRole(`${ROLE}`, authorized.address);
+        expect(
+          await instance.callStatic['hasRole(bytes32,address)'](
+            `${ROLE}`,
+            authorized.address,
+          ),
+        ).to.equal(true);
+      });
+      it('emits RoleGranted event', async function () {
+        await expect(
+          instance.connect(admin).grantRole(`${ROLE}`, authorized.address),
+        ).to.emit(instance, 'RoleGranted');
+      });
+      it('accounts can be granted a role multiple times', async function () {
+        await instance.connect(admin).grantRole(`${ROLE}`, authorized.address);
+        const trx = await instance
+          .connect(admin)
+          .grantRole(`${ROLE}`, authorized.address);
+        const receipt = await trx.wait();
+        expect(receipt.events?.length).to.equal(0);
+      });
+    });
+    describe('#revokeRole(bytes32, address)', function () {
+      beforeEach(async function () {
+        this.receipt = await instance
+          .connect(admin)
+          .grantRole(`${ROLE}`, authorized.address);
+      });
+      it('admin can revoke role', async function () {
+        await instance.connect(admin).revokeRole(`${ROLE}`, authorized.address);
+        expect(
+          await instance.callStatic['hasRole(bytes32,address)'](
+            `${ROLE}`,
+            authorized.address,
+          ),
+        ).to.equal(false);
+      });
+    });
+    describe('#renounceRole(bytes32, address)', function () {
+      beforeEach(async function () {
+        this.receipt = await instance
+          .connect(admin)
+          .grantRole(`${ROLE}`, authorized.address);
+      });
+      it('bearer can renounce role', async function () {
+        await expect(
+          instance
+            .connect(authorized)
+            .renounceRole(`${ROLE}`, authorized.address),
+        )
+          .to.emit(instance, 'RoleRevoked')
+          .withArgs(`${ROLE}`, authorized.address, authorized.address);
+
+        expect(
+          await instance.callStatic['hasRole(bytes32,address)'](
+            `${ROLE}`,
+            authorized.address,
+          ),
+        ).to.equal(false);
+      });
+      it('a role can be renounced multiple times', async function () {
+        await instance
+          .connect(authorized)
+          .renounceRole(`${ROLE}`, authorized.address);
+
+        const trx = await instance
+          .connect(authorized)
+          .renounceRole(`${ROLE}`, authorized.address);
+        const receipt = await trx.wait();
+        expect(receipt.events?.length).to.equal(0);
+      });
+    });
+    describe('reverts if', function () {
```

Comment:
`describe('reverts if')` should be nested within the function they apply to; 

eg `renounceRole` should revert if `account !== msg.sender`

Reply:
its the same method as Ownable.behavior.ts

Reply:
@iam-dev In `Ownable.behavior.ts` the revert statements are nested within the function they pertain to:
https://github.com/solidstate-network/solidstate-solidity/blob/master/spec/access/Ownable.behavior.ts

Similarly:
https://github.com/solidstate-network/solidstate-solidity/blob/9e5451f7a83a779028dc8f3ea8886fb59f1932e9/spec/token/ERC20/ERC20Base.behavior.ts#L161